### PR TITLE
Audit Plugin Work

### PR DESCRIPTION
**Pull request by BeAllAround**
Plugin to intercept and record a certain group of seneca messages specified by the user.

### CODE REVIEW

File: src/audit.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      // TODO: Do we capture prior as well?
+      if ( (properties = intercepted.find(msg)) 
+        && null == meta.prior) {
+        let reducedMsg = {}
+        // console.log( properties, msg )
+        
+        const { include, exclude } = properties
+        reducedMsg = Object.entries(msg).reduce((acc: any, pair: any) => {
+          const [ key, value ] = pair
+          if (null != include[key] && null == exclude[key]) {
```
Comment (zackbraksa) [inline src/audit.ts]: Does it make sense to handle the wildcard case? For example if I define `intercept` like this: 

```
intercept: {
        'aim:user': { include: '*', exclude: [] },
        'aim:upload': { include: '*', exclude: [] }
}
```

Comment (BeAllAround) [reply src/audit.ts]: Sure it does. Will do :+1:

Comment (zackbraksa) [reply src/audit.ts]: Awesome! Thanks 🙏

**Comment by zackbraksa** [comment]
Tested this out, and it's working smoothly. Nice work! 👍

**Pull request merged**